packages/js-core/src/shared/utils.ts

```diff
@@ -34,3 +35,34 @@ export const getDefaultLanguageCode = (survey: TSurvey) => {
   });
   if (defaultSurveyLanguage) return defaultSurveyLanguage.language.code;
 };
+
+export const handleHiddenFields = (
+  hiddenFieldsConfig: TSurvey["hiddenFields"],
+  hiddenFields: TTrackProperties["hiddenFields"]
+): THiddenFieldValue => {
+  const { enabled: enabledHiddenFields, fieldIds: hiddenFieldIds } = hiddenFieldsConfig || {};
+
+  let hiddenFieldsObject: THiddenFieldValue = {};
+
+  if (!enabledHiddenFields) {
+    console.warn("Hidden fields are not enabled for this survey");
```

Comment:
in js-core please use the logger if something should be logged to the user. when deployed formbricks shouldn't log information to all endusers. It also will be pretty common to have hidden fields in the track method but not in the survey since an action can track multiple actions and not all of them might have hidden fields.